src/platforms/web/compiler/directives/model.js

```diff
@@ -4,6 +4,7 @@ import config from 'core/config'
 import { addHandler, addProp, getBindingAttr } from 'compiler/helpers'
 import { genComponentModel, genAssignmentCode } from 'compiler/directives/model'
 
+const ternaryTextInputRE = /^[^?]{1,}\?\s*('|"|`)(?:text|number|password|search|email|tel|url)\1\s*:\s*('|"|`)(?:text|number|password|search|email|tel|url)\2\s*$/
```

Comment:
Oh, yeah, we could use 
```
/^[^?]{1,}\?\s*('|"|`)([a-z]*)\1\s*:\s*('|"|`)([a-z]*)\3\s*$/
```
and just checks `[2]` and `[4]` with `makeMap`. I'll add this changes to #6344 instead because `isTextInputType` is moved over there.